core/domain/story_services.py

```diff
@@ -65,6 +67,69 @@ def _migrate_story_contents_to_latest_schema(versioned_story_contents):
         story_schema_version += 1
 
 
+def create_new_story_exploration_link(story_id, exploration_id):
+    """Creates a new StoryExplorationLink model.
+
+    Args:
+        story_id: str. ID of the story linked to the exploration.
+        exploration_id: str. ID of the exploration linked to the story.
+    """
+    story_exploration_link_model = (
+        story_models.StoryExplorationLinkModel.create(story_id, exploration_id))
+    story_exploration_link_model.put()
+
+
+def get_pretests_for_exploration(exploration_id):
+    """Checks if an exploration is linked to a story and returns prerequisite
+    skills if so.
+
+    Args:
+        exploration_id: str. ID of the exploration.
+
+    Returns:
+        list(Question)|None. The list of questions which are the pretests for
+            the exploration, if it is linked to a story, else None.
+    """
+    story_id = (
+        story_models.StoryExplorationLinkModel.get_story_id_linked_to_exploration( #pylint: disable=line-too-long
+            exploration_id))
+
+    if story_id is None:
+        return None
+
+    story = get_story_by_id(story_id)
+    prerequisite_skill_ids = None
+    for node in story.story_contents.nodes:
+        if node.exploration_id == exploration_id:
+            prerequisite_skill_ids = node.prerequisite_skill_ids
+            break
+
+    if prerequisite_skill_ids is None:
+        return None
+
+    question_ids, _ = (
```

Comment:
Shouldn't there be some limit to the number of pre-test questions?

We should at the very least expose them in batches rather than sending every question to the frontend...

Reply:
We are. If cursor value passed is null, only 10 questions will be fetched,  defined in feconf.

Reply:
Hm. Actually, I think you should have an explicit constant for this. It should be tunable independently of other constants.

Reply:
Done.